### DISCUSSION TITLE
Correct way to deploy site

### DISCUSSION THREAD

**Discussion by jollydodo** [thuliteio/doks#1130] [Q&A]
Hey,

This is going to be a noob question, but I'm wondering what the correct way is to deploy a Hyas/doks website. For normal Hugo websites I would just copy the project files to Github with a Netlify.toml file and then Netlify would sort everything out (Hugo server dependencies are not stored in this folder). However, I first developed the website locally using the npm install method and this downloads loads of dependencies to the same folder which gets quite big (node_modules) and means you have to mess around with Github Large File Transfer. So I'm wondering if I'm taking the right approach, because this seems to be counter-intuitive. So do I really need to copy the complete folder to Github or am I approaching this the wrong way? 

Thank you for your patience. I'm so close to finishing the website ;) 
Kind regards

**Comment by h-enk** [comment]
> For normal Hugo websites I would just copy the project files to Github with a Netlify.toml file and then Netlify would sort everything out [..]

Doks (and Hyas) just work the same /alike.

The `node_modules` folder is only needed at build time — you should _not_ upload this folder to GitHub (as specified in the default [.gitignore](https://github.com/gethyas/doks/blob/main/.gitignore#L6)). However, you _do_ need to upload the `package.json` (and `package-lock.json`) files.  Netlify (or any other hosting provider) then knows to first install the project's dependencies and next run the `build` command (as specified in `package.json`) — a default [`netlify.toml`](https://github.com/gethyas/doks/blob/main/netlify.toml) is also provided.

[Learn more about dependencies](https://docs.gethyas.com/concepts/dependencies/) in the Hyas docs.

So, just go ahead as you normally would 😆
